Sources/OpenTelemetrySdk/Trace/SpanProcessors/BatchSpanProcessor.swift

```diff
@@ -109,33 +109,33 @@ private class BatchWorker: Thread {
                 spansCopy = spanList
                 spanList.removeAll()
                 cond.unlock()
-                self.exportBatch(spanList: spansCopy)
+                self.exportBatch(spanList: spansCopy, explicitTimeout: nil)
             }
         } while true
     }
 
     func shutdown() {
-        forceFlush()
+        forceFlush(explicitTimeout: nil)
         spanExporter.shutdown()
     }
 
-    public func forceFlush() {
+    public func forceFlush(explicitTimeout: TimeInterval?) {
         var spansCopy: [ReadableSpan]
         cond.lock()
         spansCopy = spanList
         spanList.removeAll()
         cond.unlock()
         // Execute the batch export outside the synchronized to not block all producers.
-        exportBatch(spanList: spansCopy)
+        exportBatch(spanList: spansCopy, explicitTimeout: explicitTimeout)
     }
 
-    private func exportBatch(spanList: [ReadableSpan]) {
+    private func exportBatch(spanList: [ReadableSpan], explicitTimeout: TimeInterval?) {
         let exportOperation = BlockOperation { [weak self] in
             self?.exportAction(spanList: spanList)
         }
         let timeoutTimer = DispatchSource.makeTimerSource(queue: DispatchQueue.global())
-        timeoutTimer.setEventHandler{ exportOperation.cancel() }
-        timeoutTimer.schedule(deadline: .now() + .milliseconds(Int(exportTimeout.toMilliseconds)), leeway: .milliseconds(1))
+        timeoutTimer.setEventHandler { exportOperation.cancel() }
+        timeoutTimer.schedule(deadline: .now() + .milliseconds(Int(explicitTimeout?.toMilliseconds ?? exportTimeout.toMilliseconds)), leeway: .milliseconds(1))
```

Comment:
is timeout that is higher than exportTimoue make sense?

Reply:
Yes, it really doesn't make much sense, it will use the lower of them